supervisor/stub/filesystem.c

```diff
@@ -78,3 +73,32 @@ void filesystem_set_concurrent_write_protection(fs_user_mount_t *vfs, bool concu
 bool filesystem_present(void) {
     return false;
 }
+
+bool filesystem_lock(fs_user_mount_t *fs_mount) {
```

Comment:
I'm confused about the semantics of this. It doesn't seem to be guaranteeing exclusive access. If the filesystem is not locked, then you attempt to lock it, and if that fails you return `false`. If you succeed in locking it, then you increment the lock count and return `true`. But if it's already locked, then you also return `true`. Wouldn't you want to return `false` if the lock count is already > 0? And it's not guarded by interrupt guards. Is that an issue?

Reply:
This is copied from the shared implementation. The filesystem locking is for a shared blockdev lock. I'm happy to rename if you can think of something better. https://github.com/adafruit/circuitpython/blob/main/supervisor/filesystem.h#L51-L53